### DISCUSSION TITLE
动态库dll中 QCefContext/QCefView等资源该如何释放

### DISCUSSION THREAD

**Discussion by HHcamel** [CefView/QCefView#111] [Q&A] [answered]
你好，
    我在dll中，创建QApplication实例，并且初始化了QCefView相关资源，并且能够正常显示，但是在UnloadLibrary时，由于QCefView等资源释放导致 程序崩溃或卡死。 
此问题有无避免的方法， 请不吝赐教！

**Comment by tishion** [comment, marked as answer]
QCefContext跟QApplication一样的，目前的设计是需要跟随程序的生命周期，不支持动态的unload。

**Reply by tishion** [reply]
为什么要卸载这个小dll呢，运行过程中只要销毁所有的QCefViews实例CEF相关的资源就会释放了。如果是因为多进程原因，你可以添加
```cpp
config.addCommandLineSwitch("single-process");
```
就可以使用单进程模式。

**Reply by HHcamel** [reply]
QApplication 和 QCefView相关资源都是在dll里初始化的，exe程序是动态加载该dll，但是在动态卸载dll时，由于dll中QCefView资源释放导致程序崩溃。
exe程序可以理解为时非 Qt写的程序，所以需要加载封装QCefView的Dll
